### PR TITLE
look for invalid names for pubmed queries

### PR DESCRIPTION
**Pull request by peetucket**
## Why was this change made?

Code changes for #1546 

- if an author has a name that includes "Or", "And", or "Not" (case insensitive, either first or last), that name will be ignored when constructing the query ... this means a user could in theory have a second "alternate identity" name that is valid, and we would still use it
- if there are no valid names available, the entire query will be marked as invalid and thus not run, producing no results
- while I was making this change, I extended our existing check for invalid names (currently only first name must have at least one character) to ensure the last name also has to have at least one valid character (eg. a last name of blank or null will also be marked as an invalid name)
- reorganize spec to put all of the tests related to the `valid?` method in a describe block

## How was this change tested?

Added new tests

### CODE REVIEW

File: spec/lib/pubmed/query_author_spec.rb [hunk 1]
```diff
@@ -195,4 +167,70 @@
...
+        expect(query_or_author.pmids).to be_empty
+      end
+
+      it 'indicates that a name with "Not" as a last name is not a valid query and returns no pmids' do
+        expect(query_not_author).not_to be_valid
+        expect(query_not_author.pmids).to be_empty
+      end
+    end
+  end
 end
```
Comment (peetucket) [inline spec/lib/pubmed/query_author_spec.rb:236]: no tests were eliminated here, just moved to a describe block, along with adding new tests verifying that names that include "Or" and "Not" are ignored, as well as nil last names

**Review by ndushay — APPROVED** [review]
